ios/Video/RCTVideo.m

```diff
@@ -350,35 +350,52 @@ - (void)setSrc:(NSDictionary *)source
 
     // perform on next run loop, otherwise other passed react-props may not be set
     [self playerItemForSource:source withCallback:^(AVPlayerItem * playerItem) {
-      _playerItem = playerItem;
-      [self addPlayerItemObservers];
-      [self setFilter:_filterName];
-      [self setMaxBitRate:_maxBitRate];
-      
-      [_player pause];
-        
-      if (_playbackRateObserverRegistered) {
+        if (self->_rctVideoDelegate == nil ||
+            ![self->_rctVideoDelegate respondsToSelector:@selector(willSetupPlayerItem:forSource:)] ||
+            ![self->_rctVideoDelegate willSetupPlayerItem:playerItem forSource:source])
```

Comment:
I don't really get this IF-statement. If `willSetupPlayerItem:playerItem` returns true you don't want to call `setupPlayerItem:playerItem`?

Reply:
No, it's like:
- if no delegate
- or delegate doesn't have method
- or delegate's method returns false

Then call the usual code

Reply:
Sure, that's what I wrote (but other way around). Are you doing the actual set up in `willSetupPlayerItem:playerItem`? I thought `willSetupPlayerItem:playerItem` was just a callback letting the delegate know it was about to set a new playerItem...

Shouldn't `setupPlayerItem` always be called? Otherwise we don't set up all the observers. Looks like `onVideoLoadStart` will not be called if we use your delegate method?

Can you please elaborate on what it is you are doing here, instead of me guessing :)

Reply:
Oh! indeed @jenshandersson ! It makes sense now!

It seems that the problem is to find the best writing.

I wrote that method to be read like "Will the `delegate` setup the `player`?" If `NO`, then `RCTVideo` will set up it.

Google IMA requires me to make a network request BEFORE having the real `AVPlayerItem` and `AVPlayer`. So I need to "delay" the player setup.

You can see it here:
- [RCGoogle IMA - Checks if has all requirements for ad playback](https://github.com/tedconf/react-native-google-ima/blob/42346b20eebfbc8d4c75420cf3a65b9697e769ca/ios/RNGoogleIMA.m#L96)
- [RCGoogle IMA - Request the ads](https://github.com/tedconf/react-native-google-ima/blob/42346b20eebfbc8d4c75420cf3a65b9697e769ca/ios/RNGoogleIMA.m#L146)
- [RCGoogle IMA - Set up player when all AD content is loaded](https://github.com/tedconf/react-native-google-ima/blob/42346b20eebfbc8d4c75420cf3a65b9697e769ca/ios/RNGoogleIMA.m#L181)
  - I don't like calling line 184 but I can't find a better way yet. Maybe if we force calling it in [RCTVideo.m#370](https://github.com/react-native-community/react-native-video/blob/5289def3ba9921c3f1d9c8adfe2be6578e010225/ios/Video/RCTVideo.m#L370) but I'm not sure the side-effects of it.

Maybe rename to `should` instead? So the `delegate` will tell if `RCTVideo` should set it up?

Reply:
Ok I get it now, but would make more sense to call it `should`.

Reply:
My reading is `will (the delegate) setup the player (later)?`, but can also be read like `delegate, should (RCTVideo) setup the player (now)?` indeed... 

Will make the change